pxr/base/arch/pragmas.h

```diff
@@ -51,7 +51,7 @@
     #define ARCH_PRAGMA_STRINGOP_OVERFLOW \
         _Pragma("GCC diagnostic ignored \"-Wstringop-overflow=\"")
 
-#elif defined(ARCH_COMPILER_CLANG)
+#elif (defined(ARCH_COMPILER_CLANG) && !defined(ARCH_COMPILER_MSVC))
```

Comment:
I tried your patch, but this doesn't look right.
Because with `clang-cl` ver. 21, I get a lot of
```c
 warning: pragma diagnostic pop could not pop, no matching push [-Wunknown-pragmas]
```

Reply:
@gvanem The pragma warning you're seeing with `gf/numericCast.h` is a known issue in the dev branch and will be fixed shortly.